pkg/gui/main_panel.go

```diff
@@ -39,7 +51,25 @@ func (gui *Gui) scrollRightMain(g *gocui.Gui, v *gocui.View) error {
 	mainView := gui.getMainView()
 	ox, oy := mainView.Origin()
 
-	return mainView.SetOrigin(ox+gui.Config.UserConfig.Gui.ScrollHeight, oy)
+	content := mainView.ViewBufferLines()
+	var largestNumberOfCharacters int
+	for _, txt := range content {
+		if len(txt) > largestNumberOfCharacters {
+			largestNumberOfCharacters = len(txt)
+		}
+	}
+
+	scrollCount := gui.Config.UserConfig.Gui.ScrollHeight
+	sizeX, _ := mainView.Size()
+	if ox+sizeX >= largestNumberOfCharacters {
+		if largestNumberOfCharacters > sizeX {
```

Comment:
I'm not sure why we need this extra condition here. I would have thought that if ` ox+sizeX >= largestNumberOfCharacters` then we could just `return nil`

Reply:
Yes, thank you for pointing that out. This is so much better.